src/client.rs

```diff
@@ -10,44 +10,59 @@ use bucket::Bucket;
 
 use utils::unwrap_collection_records;
 
-
-/// Client for the Kinto HTTP API.
-#[derive(Debug)]
-pub struct KintoClient {
+/// Configuration of the Kinto endpoint.
+#[derive(Debug, Clone)]
+pub struct KintoConfig {
     pub server_url: String,
-    pub http_client: client::Client,
     pub auth: Option<Authorization<Basic>>,
 }
 
+impl KintoConfig {
+    pub fn new(server_url: String, auth: Option<Authorization<Basic>>) -> KintoConfig {
+        KintoConfig {
+            server_url: server_url,
+            auth: auth,
+        }
+    }
 
-impl KintoClient {
-    /// Create a client.
-    pub fn new(server_url: String, auth: Option<Authorization<Basic>>) -> KintoClient {
-
+    pub fn http_client(&self) -> client::Client {
         // Build an SSL connector
         let ssl = NativeTlsClient::new().unwrap();
         let connector = HttpsConnector::new(ssl);
 
         // Build a HTTP Client with TLS support.
-        let client = client::Client::with_connector(connector);
+        client::Client::with_connector(connector)
+    }
+}
+
+/// Client for the Kinto HTTP API.
+#[derive(Debug)]
+pub struct KintoClient {
+    pub http_client: client::Client,
+    config: KintoConfig,
+}
+
+
+impl KintoClient {
+    /// Create a client.
+    pub fn new(config: KintoConfig) -> KintoClient {
 
         KintoClient {
-            server_url: server_url,
-            http_client: client,
-            auth: auth,
+            config: config.clone(),
```

Comment:
Do we need to `clone` the config? It seems like the KintoClient takes ownership of it.